### DISCUSSION TITLE
Not reload in dev with freebsd

### DISCUSSION THREAD

**Discussion by josejachuf** [quasarframework/quasar#11124] [General - Components / Directives / etc] [answered]
Hi, when I'm developed in FreeBSD, the recharge of the modifications I'm doing is doing. I have to stop the dev and restart it again to make the changes.

What can I see or touch to solve this?

All the info of what I have installed is below 

> % quasar info
> 
> Operating System - FreeBSD(13.0-RELEASE-p4) - freebsd/x64
> NodeJs - 14.18.1
> 
> Global packages
>   NPM - 6.14.8
>   yarn - 1.22.11
>   @quasar/cli - 1.2.2
>   @quasar/icongenie - 2.4.0
>   cordova - Not installed
> 
> Important local packages
>   quasar - 2.2.1 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
>   @quasar/app - 3.1.7 -- Quasar Framework local CLI
>   @quasar/extras - 1.11.3 -- Quasar Framework fonts, icons and animations
>   eslint-plugin-quasar - Not installed
>   vue - 3.2.20 -- The progressive JavaScript framework for buiding modern web UI.
>   vue-router - 4.0.12
>   vuex - 4.0.2 -- state management for Vue.js
>   electron - Not installed
>   electron-packager - Not installed
>   electron-builder - Not installed
>   @babel/core - 7.15.8 -- Babel compiler core.
>   webpack - 5.59.1 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
>   webpack-dev-server - 4.3.1 -- Serves a webpack app. Updates the browser on changes.
>   workbox-webpack-plugin - Not installed
>   register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
>   typescript - 4.4.2 -- TypeScript is a language for application scale JavaScript development
>   @capacitor/core - Not installed
>   @capacitor/cli - Not installed
>   @capacitor/android - Not installed
>   @capacitor/ios - Not installed
> 
> Quasar App Extensions
>   *None installed*
>

**Comment by josejachuf** [comment]
In the only case that rebuild the App is when I modify the file quasar.conf.js

**Reply by hawkeye64** [reply]
Have you increased your watch files for the OS? Sometimes there's not many and your editor can use them up quickly. 

Read this and see if it helps: https://dev.to/stephencweiss/running-vscode-on-linux-hitting-my-watcher-limit-and-what-to-do-about-it-52i0

**Reply by josejachuf** [reply]
Hello, thanks for the answer.
In FreeBSD I do not know how it is or I do not have it, what I have is the following:

> % sysctl kern.maxfiles kern.maxfilesperproc
> kern.maxfiles: 647044
> kern.maxfilesperproc: 582336

Only an observation, what I mentioned happens from the console, not necessarily since vscode 
Jose

**Reply by josejachuf** [reply]
In a Vue3 project (not quasar) the reload works fine

**Reply by hawkeye64** [reply]
I'm sorry. I am running out of ideas. I use Ubuntu and don't have this issue.

Are you using Typescript? Or `<script setup>`?

**Reply by josejachuf** [reply]
In Linux Mint, which I also use there are no problems. And in earlier versions of Quasar in FreeBSD, there were no problems either.

Thanks anyway

**Comment by josejachuf** [comment, marked as answer]
Hello @hawkeye64  I had told you that it was not a problem with vs-code, but it seems that if.
If I run from a console "quasar dev" and I edit with vs-code, fail, no changes are recharged, as quasar executed it from the console I thought it was not the problem.

Now I'm using Neovim as editor and everything works fine (at least most of the time)

**Comment by schlomie** [comment]
In `quasar.conf.js` Add the following to the `build` block

``` diff
build: {
  chainWebpack (chain) {
    chain.plugin('eslint-webpack-plugin').use(ESLintPlugin, [{ extensions: [ 'js', 'vue' ] }])
  },
+ extendWebpack(cfg) {
+   cfg.watchOptions = {
+     aggregateTimeout: 200,
+     poll: 1000,
+   }
+ }
}
```

Not quite as efficient, but this does work on FreeBSD.

For reference: [Docker and WSL Issues with HMR](https://quasar.dev/quasar-cli-webpack/quasar-config-js#docker-and-wsl-issues-with-hmr)

**Reply by josejachuf** [reply]
It was a problem with vscode, now use Neovim and works fine
